metadata/metadata.go

```diff
@@ -482,6 +511,74 @@ func (resource *sourceVariantResource) Update(lookup ResourceLookup, updateRes R
 	return nil
 }
 
+func (resource *sourceVariantResource) IsEquivalent(other ResourceVariant) (bool, error) {
+	/**
+	Key Fields for a SourceVariant are
+	- Name
+	- Definition
+	- Owner
+	- Provider
+	*/
+	otherVariant, ok := other.(*sourceVariantResource)
+	if !ok {
+		return false, nil
+	}
+
+	thisProto := resource.serialized
+	otherProto := otherVariant.serialized
+
+	isDefinitionEqual := false
+	var err error
+	switch thisDef := thisProto.Definition.(type) {
+	case *pb.SourceVariant_Transformation:
+		if otherDef, ok := otherProto.Definition.(*pb.SourceVariant_Transformation); ok {
+			isDefinitionEqual, err = isProtoDefinitionEqual(thisDef, otherDef)
+			if err != nil {
+				return false, fmt.Errorf("error comparing proto messages: %v", err)
```

Comment:
from the user's perspective, would the error message make sense? would they know what is `photo messages`?

Reply:
no, but what else could you say?

Reply:
`error comparing two resources: %v`?

Reply:
bump

Reply:
let me try